client/ayon_nuke/api/lib.py

```diff
@@ -810,89 +812,24 @@ def check_inventory_versions():
-    repre_entities = ayon_api.get_representations(
-        project_name,
-        representation_ids=repre_ids,
-        fields={"id", "versionId"}
-    )
-    # Store representations by id and collect version ids
-    repre_entities_by_id = {}
-    version_ids = set()
-    for repre_entity in repre_entities:
-        # Use stringed representation id to match value in containers
-        repre_id = repre_entity["id"]
-        repre_entities_by_id[repre_id] = repre_entity
-        version_ids.add(repre_entity["versionId"])
-
-    version_entities = ayon_api.get_versions(
-        project_name,
-        version_ids=version_ids,
-        fields={"id", "version", "productId"},
-    )
-    # Store versions by id and collect product ids
-    version_entities_by_id = {}
-    product_ids = set()
-    for version_entity in version_entities:
-        version_entities_by_id[version_entity["id"]] = version_entity
-        product_ids.add(version_entity["productId"])
-
-    # Query last versions based on product ids
-    last_versions_by_product_id = ayon_api.get_last_versions(
-        project_name, product_ids=product_ids, fields={"id", "productId"}
-    )
-
-    # Loop through collected container nodes and their representation ids
-    for item in node_with_repre_id:
-        # Some python versions of nuke can't unfold tuple in for loop
-        node, repre_id = item
-        repre_entity = repre_entities_by_id.get(repre_id)
-        # Failsafe for not finding the representation.
-        if not repre_entity:
-            log.warning((
-                "Could not find the representation on node \"{}\""
-            ).format(node.name()))
-            continue
 
-        version_id = repre_entity["versionId"]
-        version_entity = version_entities_by_id.get(version_id)
-        if not version_entity:
-            log.warning((
-                "Could not find the version on node \"{}\""
-            ).format(node.name()))
-            continue
+    category_colors = {
+        "latest": "0x4ecd25ff",
+        "outdated": "0xd84f20ff",
+        "invalid": "0xff0000ff",
+        "not_found": "0xffff00ff",
+    }
 
-        # Get last version based on product id
-        product_id = version_entity["productId"]
-        last_version = last_versions_by_product_id[product_id]
-        # Check if last version is same as current version
-        if last_version["id"] == version_entity["id"]:
-            color_value = "0x4ecd25ff"
-        else:
-            color_value = "0xd84f20ff"
-        node["tile_color"].setValue(int(color_value, 16))
+    filtered_containers = filter_containers(containers, project_name)
+    for category, containers in filtered_containers._asdict().items():
+        container_color = category_colors.get(category) or category_colors["not_found"]
+        for container in containers:
+            container["node"]["tile_color"].setValue(
+                int(container_color, 16)
+            )
```

Comment:
This may be nicer?
```suggestion
    for category, containers in filtered_containers._asdict().items():
        color = category_colors.get(category) or category_colors["not_found"]
        color = int(color, 16)  # convert hex to nuke tile color int
        for container in containers:
            container["node"]["tile_color"].setValue(color)
```

I'm just curious why you ended up doing the `or category_colors["not_found"]` fallback. It feels a bit as if it may be hiding another issue accidentally?

I think, if it's an unknown category for whatever reason we should just ignore it.

```suggestion
    for category, containers in filtered_containers._asdict().items():
        if category not in category_colors:
            continue
        color = category_colors[category]
        color = int(color, 16)  # convert hex to nuke tile color int
        for container in containers:
            container["node"]["tile_color"].setValue(color)
```

Reply:
> I'm just curious why you ended up doing the or category_colors["not_found"] fallback. It feels a bit as if it may be hiding another issue accidentally?

Yeah... just being cautious of possible changes on the API that could rephrase the categories or any other issues so it doesn't try to apply `None`. Happy to ignore those cases, perhaps even adding a warning?